.github/workflows/pull-request-ci.yml

```diff
@@ -53,19 +53,19 @@ jobs:
         uses: actions/cache@v2
         with:
           path: external
-          key: ${{ runner.os }}-external-${{ hashFiles('src/Radium-Engine/external/**/CMakeLists.txt') }}
+          key: ${{ matrix.config }}-external-${{ hashFiles('src/Radium-Engine/external/**/CMakeLists.txt') }}
```

Comment:
well this solution do not work, the key is 
`Object-external-c20c9af153db4d90af43dee80421f2b56ba8b04b59beb31bf701b2054d21790a`
so matrix.config becomes "Object" in the key.
Maybe it's simpler to use the "version hack" if I can name it like that.
By the way next cache retrieval with this key may mess up the different build due to key collision for all OS.